### ISSUE TITLE
S3 HOSTNAME_EXTERNAL does not change the returned URL for S3 items

### ISSUE THREAD

**Issue by monsonjeremy** [localstack/localstack#398]
Currently when setting the environment variable HOSTNAME_EXTERNAL, after uploading an item to S3 the Location attribute returned in the response has the URL "http://localstack:" instead of being assigned the HOSTNAME_EXTERNAL environment variable value.

To add some context:
I want to upload an image to S3 and then use the returned URL to add an image in my front-end. Therefore, I want the URL to be localhost:5001 (i've exposed this port from my docker-compose and s3 is on port 5001) so that I can access this image from the browser on my local.

**Comment by neilsh** [comment]
I think that HOSTNAME_EXTERNAL is currently only used by SQS, but I think hosting S3 uploads on localhost is a great idea. I don't believe there's any mechanism to currently access S3 uploads via http, but I could be mistaken.

**Comment by monsonjeremy** [comment]
I am able to actually access the image in the browser if I change the returned URL to have "localhost" instead of "localstack" so it's just a matter of changing the URL returned in the response of the upload to s3. Currently the response has a location with url like "...localstack:..." it needs to be "....localhost:...."

**Comment by jk2l** [comment]
+1 i want to write an integration test that simulate s3 object creation event handling. but i am having trouble setup the tests with custom endpoint within docker

**Comment by jk2l** [comment]
```
#!/usr/bin/env bash

set +e

CURRENT_DIR=$(dirname $(realpath $0))

# Make sample environments for development
awslocal s3 mb s3://batch_tasks
awslocal s3 mb s3://images
awslocal s3 cp "$CURRENT_DIR/fixtures/sample-image.jpeg" s3://images/sample-image.jpeg

awslocal sqs create-queue --queue-name=s3_input_queue

awslocal sns create-topic --name=s3_input
awslocal sns subscribe --topic-arn arn:aws:sns:us-east-1:123456789012:s3_input --protocol sqs --notification-endpoint http://aws.docker.localhost:4576/queue/s3_input_queue

awslocal s3api put-bucket-notification-configuration --bucket batch_tasks --notification-configuration "TopicConfigurations={TopicArn=arn:aws:sns:us-east-1:123456789012:s3_input,Events=[s3:ObjectCreated:*]}"`
```
currently my endpoint have to be localhost:4572 for docker, even it should be custom dns "aws.docker.localhost". The reason of using custom endpoint is my localstack and integration tests run with different container, with network alias pointing aws.docker.localhost to localstack container. the "localhost" inside integration test container won't be able to find localstack container unless i bridge two container to share same network bridge.

**Comment by mdere-unbound** [comment]
Yup running the same issue here - my workaround was to do some string manipulation on the domain name provided back to me and change it to what I want, just a little more code - but wish we can configure it.

**Comment by whummer** [comment]
Should be fixed in #1171 - Location attribute is now correctly returned based on `HOSTNAME_EXTERNAL` configuration variable. Please report here and re-open this ticket if the problem persists. Thanks
